recharges/tasks.py

```diff
@@ -17,31 +93,81 @@ class Hotsocket_Login(Task):
     name = "recharges.tasks.hotsocket_login"
 
     def run(self, **kwargs):
-        """
-        Returns the token
-        """
-        l = self.get_logger(**kwargs)
-        l.info("Logging into hotsocket")
 
-        auth = {'username': settings.HOTSOCKET_API_USERNAME,
-                'password': settings.HOTSOCKET_API_PASSWORD,
-                'as_json': True}
-
-        r = requests.post("%s/login" % settings.HOTSOCKET_API_ENDPOINT,
-                          data=auth)
-        result = r.json()
+        l = self.get_logger(**kwargs)
+        login_result = request_hotsocket_login()
+        status = login_result["response"]["status"]
         # Check the result
-        if result["response"]["status"] == \
-                settings.HOTSOCKET_CODES["LOGIN_SUCCESSFUL"]:
+        if status == settings.HOTSOCKET_CODES["LOGIN_SUCCESSFUL"]:
             l.info("Successful login to hotsocket")
-            # Store the result for other tasks to use
-            token = result["response"]["token"]
-            account = Account()
-            account.token = token
-            account.save()
+            Account.objects.create(token=login_result["response"]["token"])
             return True
         else:
             l.error("Failed login to hotsocket")
             return False
 
 hotsocket_login = Hotsocket_Login()
+
+
+class Hotsocket_Process_Queue(Task):
+
+    """
+    Task to get the get all unprocessed recharges and create tasks to
+    submit them to hotsocket
+    """
+    name = "recharges.tasks.hotsocket_process_queue"
+
+    def run(self, **kwargs):
+        """
+        Returns the number of submitted requests
+        """
+        l = self.get_logger(**kwargs)
+        l.info("Looking up the unprocessed requests")
+        queued = Recharge.objects.filter(status=0)
+        for recharge in queued:
+            hotsocket_get_airtime.delay(recharge.id)
+        return "%s requests queued to Hotsocket" % queued.count()
+
+hotsocket_process_queue = Hotsocket_Process_Queue()
+
+
+class Hotsocket_Get_Airtime(Task):
+
+    """
+    Task to make hotsocket post request to load airtime, saves hotsocket ref
+    to the recharge model and update status
+    """
+    name = "recharges.tasks.hotsocket_get_airtime"
+
+    def run(self, recharge_id, **kwargs):
+        """
+        Returns the recharge model entry
+        """
+        l = self.get_logger(**kwargs)
+        recharge = get_recharge(recharge_id)
+        cell_number = recharge.msisdn
```

Comment:
Please just use `recharge.msisdn` where you're using `cell_number`. There's no value to setting it to a variable as we're not manipulating it at all.